src/conduit/components/profile.cljs

```diff
@@ -0,0 +1,44 @@
+(ns conduit.components.profile
+  (:require [rum.core :as rum]
+            [citrus.core :as citrus]
+            [conduit.mixins :as mixins]
+            [conduit.components.base :as base]))
+
+(rum/defc Banner
+  [{:keys [image username bio following]} current-user on-follow on-unfollow]
+  (let [profile-owner? (= (:username current-user) username)]
+    [:.col-xs-12.col-md-10.offset-md-1
+     (when-not (empty? image) [:img.user-img {:src image}])
+     [:h4 username]
+     [:p bio]
+     [:.pull-xs-right
+      (when current-user
+        (if-not profile-owner?
+          (base/FollowButton username following on-follow on-unfollow)
+          (base/Button {:href (str "#/settings")
+                        :icon :gear-a
+                        :type :secondary}
+                       "Edit Profile Settings")))]]))
+
+(rum/defc Profile < rum/reactive
+  (mixins/dispatch-on-mount
+    (fn [_ _ {:keys [id]} {:keys [token]}]
+      {:profile [:load-profile id token]}))
+  {:will-unmount
+   (fn [{[r] :rum/args :as state}]
+     (citrus/dispatch! r :profile :init)
```

Comment:
why do we need this dispatch here?

Reply:
To clear the last viewed profile so that when next time entering another user profile page, it doesn't get mixed with the previously loaded profile. But it's probably unnecessary as we are checking the loading state already and hiding the view when loading state is not ready.